src/components/listgroup/ListGroupItem.js

```diff
@@ -1,10 +1,37 @@
+import React from 'react';
 import PropTypes from 'prop-types';
 import {ListGroupItem as RSListGroupItem} from 'reactstrap';
+import Link from '../../private/Link';
 
-const ListGroupItem = props => {
-  const {children, ...otherProps} = props;
-  return <RSListGroupItem {...otherProps}>{children}</RSListGroupItem>;
-};
+class ListGroupItem extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.incrementClicks = this.incrementClicks.bind(this);
+  }
+
+  incrementClicks() {
+    if (!this.props.disabled) {
+      if (this.props.setProps) {
+        this.props.setProps({
+          n_clicks: this.props.n_clicks + 1,
+          n_clicks_timestamp: Date.now()
+        });
+      }
+    }
+  }
+
+  render() {
+    let {children, ...otherProps} = this.props;
+    const {href, disabled} = this.props;
+    otherProps[href ? 'preOnClick' : 'onClick'] = this.incrementClicks;
+    return (
+      <RSListGroupItem tag={href && !disabled ? Link : 'li'} {...otherProps}>
```

Comment:
The associativity between `&&` and `?` are not obvious. Would it be clearer with brackets?

```
(href && !disabled) ? Link : 'li'
```

Reply:
In fact, since in PR #83 you use a temporary variable called `useLink`, is there value in using the same construct here?

Using the same syntax when code flows are similar helps a lot with maintenance.